client/src/modules/dashboard/index.jsx

```diff
@@ -59,9 +65,27 @@ class Dashboard extends Component {
 
   hideDialog = () => this.handleDialogContext(null)();
 
+  handleArchivedListsVisibility = () => {
+    this.setState(({ areArchivedListVisible }) => ({
+      areArchivedListVisible: !areArchivedListVisible
+    }));
+    this.handleArchivedListsData();
```

Comment:
this method uses `areArchivedListVisible` from state. To be sure that value will be set in state already use `setState` callback param to call this method